WooCommerce/Classes/ViewRelated/Orders/Order Filters/Date Range Filter/Custom Range Filter/DateRangeFilterViewController.swift

```diff
@@ -0,0 +1,231 @@
+
+    @IBOutlet private weak var tableView: UITableView!
+
+    // Completion callback
+    //
+    typealias Completion = (Date?, Date?) -> Void
+    private let onCompletion: Completion
+
+    private var startDate: Date?
+    private var endDate: Date?
+
+    // Indicate when the start or the end date picker is expanded
+    //
+    private var startDateExpanded: Bool = false
+    private var endDateExpanded: Bool = false
+
+    private var rows: [Row] = []
+
+    /// Init
+    ///
+    init(startDate: Date?,
+         endDate: Date?,
+         completion: @escaping Completion) {
+        self.startDate = startDate
+        self.endDate = endDate
+        onCompletion = completion
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        configureNavigationBar()
+        configureMainView()
+        configureRows()
+        configureTableView()
+    }
+}
+
+// MARK: - View Configuration
+//
+private extension DateRangeFilterViewController {
+
+    func configureNavigationBar() {
+        title = Localization.navigationBarTitle
+    }
+
+    func configureMainView() {
+        view.backgroundColor = .listBackground
+    }
+
+    func configureTableView() {
+        tableView.dataSource = self
+        tableView.delegate = self
+
+        /// Registers all of the available TableViewCells
+        ///
+        for row in Row.allCases {
+            tableView.registerNib(for: row.type)
+        }
+
+        tableView.backgroundColor = .listBackground
+        tableView.removeLastCellSeparator()
+    }
+
+    func configureRows() {
+        var tempRows: [Row] = [.startDateTitle]
+        if startDateExpanded {
+            tempRows.append(.startDatePicker)
+        }
+        tempRows.append(.endDateTitle)
+        if endDateExpanded {
+            tempRows.append(.endDatePicker)
+        }
+        rows = tempRows
+    }
+}
+
+// MARK: - UITableViewDataSource Conformance
+//
+extension DateRangeFilterViewController: UITableViewDataSource {
+
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return rows.count
+    }
+
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let row = rows[indexPath.row]
+        let cell = tableView.dequeueReusableCell(withIdentifier: row.reuseIdentifier, for: indexPath)
+        configure(cell, for: row, at: indexPath)
+        return cell
+    }
+}
+
+// MARK: - UITableViewDelegate Conformance
+//
+extension DateRangeFilterViewController: UITableViewDelegate {
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        tableView.deselectRow(at: indexPath, animated: true)
+
+        switch rows[indexPath.row] {
+        case .startDateTitle:
+            startDateExpanded = !startDateExpanded
```

Comment:
nit: another way to toggle the value that might read more naturally 😄 

```suggestion
            startDateExpanded.toggle()
```

Reply:
TIL 💯